xsnap/sources/xsnap-worker.c

```diff
@@ -331,10 +345,25 @@ int main(int argc, char* argv[])
 	}
 	xsInitializeSharedCluster();
 	if (argr) {
-		snapshot.stream = fopen(argv[argr], "rb");
-		if (snapshot.stream) {
+		char *path = argv[argr];
+		SnapshotStream stream;
+		if (path[0] == '@') {
+			int fd = atoi(path + 1);
+			int tmpfd = dup(fd);
```

Comment:
What is the `dup()` for? Does it allow us to `fclose(stream.file)` but then re-use the original fd later? Do we have a use for that, or is this a generic pattern that we *do* need for the writing side?

Reply:
We do need it for the writing side, and I figured in the future we may want the ability to reuse the load snapshot stream.